### PR TITLE
feat(block-hub): support in edgeless mode

### PR DESCRIPTION
**Pull request by fundon**
Related to #1705 

* [x] Support dragging blocks from block hub into existing frames.
* [x] Support dragging blocks from block hub to blank area.
* [x] tests

### CODE REVIEW

**Comment by fundon** [comment]
@doodlewind In which mode is `block-hub` enabled, dragged and dropped?

**Review by doodlewind — COMMENTED** [review]
Found the only failing case is the regression in the `drag database` case. Can @zqran help taking a look?

**Comment by zqran** [comment]
> Found the only failing case is the regression in the `drag database` case. Can @zqran help taking a look?

ok, let me check

File: packages/editor/src/utils/editor.ts [hunk 1]
```diff
@@ -25,53 +27,104 @@ export const createBlockHub: (
...
-    assertExists(defaultPageBlock);
-    blockHub.slots = defaultPageBlock.slots;
-    blockHub.getAllowedBlocks = () =>
-      getAllowSelectedBlocks(defaultPageBlock.model);
-  } else {
-    const edgelessPageBlock = editor.querySelector('affine-edgeless-page');
-    assertExists(edgelessPageBlock);
-    blockHub.getAllowedBlocks = () =>
-      getAllowSelectedBlocks(edgelessPageBlock.model);
-  }
```
Comment (fundon) [inline packages/editor/src/utils/editor.ts:74]: This code will only be executed once. 

May need an update subscription when switching the mode.

/cc @doodlewind

**Review by doodlewind — APPROVED** [review]
Let's land this, thanks for your hard work!

File: packages/blocks/src/page-block/edgeless/mode-controllers/default-mode.ts [hunk 2]
```diff
@@ -180,7 +179,7 @@ export class DefaultModeController extends MouseModeController<DefaultMouseMode>
   }
 
   private _tryDeleteEmptyBlocks() {
-    const emptyBlocks = this._blocks.filter(b => isEmpty(b));
+    const emptyBlocks = this._blocks.filter(b => !b.children.length);
```
Comment (alt1o) [inline packages/blocks/src/page-block/edgeless/mode-controllers/default-mode.ts:182]: This broken test `add empty text`.  Should fix the skip test `add empty text`.

**Pull request merged**